synapse/storage/data_stores/main/receipts.py

```diff
@@ -280,7 +280,7 @@ def get_all_updated_receipts_txn(txn):
                 args.append(limit)
             txn.execute(sql, args)
 
-            return (r[0:5] + (json.loads(r[5]),) for r in txn)
+            return [r[0:5] + (json.loads(r[5]),) for r in txn]
```

Comment:
should we not wrap this in an explicit `list()`, to force it from being a generator?

Reply:
isn't that the same as using `[]` ?

Reply:
oh, yes, the `()`s you replaced keep it being a generator (because it doesn't cast it to a tuple, while `[]` does cast to a list, because python is an intuitive language)

I'd personally do `list(r[0:5] + (json.loads(r[5]),) for r in txn)` to make the casting more obvious, but as-written is fine